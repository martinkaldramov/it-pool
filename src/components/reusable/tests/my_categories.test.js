import React from 'react';
import { shallow } from 'enzyme';
import MyCategories from '../my_categories';
import MockProfile from '../../../mock_data/mock_profile';

describe('MyCategories', () => {
  const wrap = shallow(<MyCategories categories={MockProfile.categories} />);

  it('should renderr successfully', () => {
    expect(wrap).toMatchSnapshot();
  });
});
