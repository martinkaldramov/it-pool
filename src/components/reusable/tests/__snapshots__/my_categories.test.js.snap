// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyCategories should renderr successfully 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MyCategories
    categories={
        Array [
            "Web Development",
            "Web & Mobile Design",
            "Scripts & Utilities",
            "Desktop Software Development",
            "Quality Assurance",
            "Frontend Web Development",
            "Network Engineering",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <MenuItem
          anchorOrigin={
                    Object {
                              "horizontal": "right",
                              "vertical": "top",
                            }
          }
          checked={false}
          desktop={false}
          disabled={false}
          focusState="none"
          insetChildren={false}
          targetOrigin={
                    Object {
                              "horizontal": "left",
                              "vertical": "top",
                            }
          }
>
          MY CATEGORIES
</MenuItem>,
        <ul>
          <li>
                    Web Development
          </li>
          <li>
                    Web & Mobile Design
          </li>
          <li>
                    Scripts & Utilities
          </li>
          <li>
                    Desktop Software Development
          </li>
          <li>
                    Quality Assurance
          </li>
          <li>
                    Frontend Web Development
          </li>
          <li>
                    Network Engineering
          </li>
</ul>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "anchorOrigin": Object {
            "horizontal": "right",
            "vertical": "top",
          },
          "checked": false,
          "children": "MY CATEGORIES",
          "desktop": false,
          "disabled": false,
          "focusState": "none",
          "insetChildren": false,
          "targetOrigin": Object {
            "horizontal": "left",
            "vertical": "top",
          },
        },
        "ref": null,
        "rendered": "MY CATEGORIES",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              Web Development
</li>,
            <li>
              Web & Mobile Design
</li>,
            <li>
              Scripts & Utilities
</li>,
            <li>
              Desktop Software Development
</li>,
            <li>
              Quality Assurance
</li>,
            <li>
              Frontend Web Development
</li>,
            <li>
              Network Engineering
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Web Development",
            },
            "ref": null,
            "rendered": "Web Development",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Web & Mobile Design",
            },
            "ref": null,
            "rendered": "Web & Mobile Design",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Scripts & Utilities",
            },
            "ref": null,
            "rendered": "Scripts & Utilities",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Desktop Software Development",
            },
            "ref": null,
            "rendered": "Desktop Software Development",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Quality Assurance",
            },
            "ref": null,
            "rendered": "Quality Assurance",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Frontend Web Development",
            },
            "ref": null,
            "rendered": "Frontend Web Development",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Network Engineering",
            },
            "ref": null,
            "rendered": "Network Engineering",
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <MenuItem
            anchorOrigin={
                        Object {
                                    "horizontal": "right",
                                    "vertical": "top",
                                  }
            }
            checked={false}
            desktop={false}
            disabled={false}
            focusState="none"
            insetChildren={false}
            targetOrigin={
                        Object {
                                    "horizontal": "left",
                                    "vertical": "top",
                                  }
            }
>
            MY CATEGORIES
</MenuItem>,
          <ul>
            <li>
                        Web Development
            </li>
            <li>
                        Web & Mobile Design
            </li>
            <li>
                        Scripts & Utilities
            </li>
            <li>
                        Desktop Software Development
            </li>
            <li>
                        Quality Assurance
            </li>
            <li>
                        Frontend Web Development
            </li>
            <li>
                        Network Engineering
            </li>
</ul>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "anchorOrigin": Object {
              "horizontal": "right",
              "vertical": "top",
            },
            "checked": false,
            "children": "MY CATEGORIES",
            "desktop": false,
            "disabled": false,
            "focusState": "none",
            "insetChildren": false,
            "targetOrigin": Object {
              "horizontal": "left",
              "vertical": "top",
            },
          },
          "ref": null,
          "rendered": "MY CATEGORIES",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                Web Development
</li>,
              <li>
                Web & Mobile Design
</li>,
              <li>
                Scripts & Utilities
</li>,
              <li>
                Desktop Software Development
</li>,
              <li>
                Quality Assurance
</li>,
              <li>
                Frontend Web Development
</li>,
              <li>
                Network Engineering
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Web Development",
              },
              "ref": null,
              "rendered": "Web Development",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Web & Mobile Design",
              },
              "ref": null,
              "rendered": "Web & Mobile Design",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Scripts & Utilities",
              },
              "ref": null,
              "rendered": "Scripts & Utilities",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Desktop Software Development",
              },
              "ref": null,
              "rendered": "Desktop Software Development",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Quality Assurance",
              },
              "ref": null,
              "rendered": "Quality Assurance",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Frontend Web Development",
              },
              "ref": null,
              "rendered": "Frontend Web Development",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Network Engineering",
              },
              "ref": null,
              "rendered": "Network Engineering",
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
